vhost-device-vsock/src/main.rs

```diff
@@ -82,8 +82,31 @@ struct VsockParam {
     socket: String,
 
     /// Unix socket to which a host-side application connects to.
-    #[arg(long, conflicts_with = "config", conflicts_with = "vm")]
-    uds_path: String,
+    #[arg(
+        long,
+        conflicts_with = "forward_cid",
+        conflicts_with = "config",
+        conflicts_with = "vm"
+    )]
+    uds_path: Option<String>,
+
+    /// The vsock CID to forward connections from guest
+    #[clap(
+        long,
+        conflicts_with = "uds_path",
+        conflicts_with = "config",
+        conflicts_with = "vm"
+    )]
+    forward_cid: Option<u32>,
+
+    /// The vsock ports to forward connections from host
+    #[clap(
+        long,
+        conflicts_with = "uds_path",
+        conflicts_with = "config",
+        conflicts_with = "vm"
+    )]
+    vsock_listen: Option<String>,
```

Comment:
Can we add a single option to cover both? I mean something like we did for `--vm`, for example `--forward cid=3,listen=1234`

Reply:
Shouldn't these options be tied to the VM's options (i.e., part of the --vm)? The uds-path option is part of the --vm and decides that the backend type is unix and when the forward-cid will be used the backend type will be vsock for that VM which makes sense, right? Separating them would be confusing IMHO and would introduce dependency on the order of the provided options like the --forward affects the previous --vm. Maybe I am misunderstanding something here. What do you think @stefano-garzarella ?

Reply:
Yep, good point, it should be part of --vm too.
What I mean, is to have a single parameter to handle `cid` and `listen`, so I think we need both --forward (used for a single vm use case), and also add `forward` handling in the `--vm`

Reply:
Do you think the separate option is strictly necessary though although the options are available as part of --vm already? I think the way it is now, can be considered good enough, no?

Reply:
> Do you think the separate option is strictly necessary though although the options are available as part of --vm already? I think the way it is now, can be considered good enough, no?

Sorry I don't get it.

We initially supported only a single vm, so `--vm` wasn't here, so we had for example `--uds-path`, then we added `--vm ...,uds-path=...`, so now we support both, so if we add a new `--foo` we also need to add `--vm ...,foo=...`. At least until we decided to deprecate `--uds-path`, and others, to support just `--vm`.

Also, please update the README, so as reviewer we can understand better how this new feature will be used.

Reply:
If you need 2 different parameters, because a single one is hard for `--vm`, so maybe better with the same prefix, something like `--forward-cid` and `--forward-listen`.

Reply:
BTW it looks like you already did it, so maybe just change the name should be fine, keeping 2 different parameters.

Reply:
oh I didn't realize a separate --uds-path option is supported. That is why it was confusing. I will look into it a bit more.

Reply:
I am a bit confused still. I only added support for the new options as part of the --vm option (as per the PR descriptio). I did not add the separate --forward option. I guess only being part of --vm should be good enough and I just need to change the 'vsock-listen' to 'forward-listen', right?

Reply:
> Sorry, I am a bit confused still. I only added support for the new options as part of the --vm option (as per the PR descriptio). I did not add the separate --forward option.

With this change at `VsockParam` you added two new options, see the output of `vhost-device-vsock --help`:
```
$ ../target/debug/vhost-device-vsock --help
A virtio-vsock device using the vhost-user protocol.

Usage: vhost-device-vsock [OPTIONS] --socket <SOCKET>
...

      --forward-cid <FORWARD_CID>
          The vsock CID to forward connections from guest
      --vsock-listen <VSOCK_LISTEN>
          The vsock ports to forward connections from host
```

>I guess only being part of --vm should be good enough and I just need to change the 'vsock-listen' to 'forward-listen', right?

Right, just to try to clear it a bit, if you add a new field (e.g. `foo`) to `VsockParam`, you're adding a new `--foo` option, if you add a new match in `parse_vm_params()`, you are supporting also `--vm foo=...`.

Since you did both (correctly), you're already supporting both `--forward-cid` and `--vm forward-cid=...`.

So you need to rename `vsock-listen` on both to be consistent.

Reply:
oh okay got it. Thanks for the clarification. I will rename both places.

Reply:
Renamed now.